### PR TITLE
try vitest coverage gh action

### PR DESCRIPTION
**Pull request by TheHaff**
So this adds coverage checks for vitest.
It is per package level and there is a way to aggregate into one with some more magic setup (I think), then we would get just one message with everything in the pr. But from experience in the client it can get infurating when some miscalculation prevents you from merging etc. So i think this is a better choice even though it posts multiple messages to the pr.
I set the coverage thresholds to whatever they were at now to get a green build but ideally the sweet spot is 80%

### CODE REVIEW

**Comment by michpohl** [comment]
How about only reporting coverage for packages with changes? Might make it a little less noisy...?

File: .github/workflows/unit-test.yml
Comment (michpohl) [inline .github/workflows/unit-test.yml:37]: I assume you don't use `working directory: './'` so you get separate reports?

File: packages/template/src/index.test.ts
Comment (michpohl) [inline packages/template/src/index.test.ts:1]: dummy non-web-dev question: What's the educational purpose of this newly added test?

**Review by michpohl — COMMENTED** [review]
I have some questions, but mostly I don't think it's a good idea to have a hardcoded block per module. So that we can move forward: It would be - for me - fine to merge like this now (given all other things are non-issues), if there was a ticket to fix this in place for next sprint, that could be referenced here.

File: packages/template/src/index.test.ts
Comment (TheHaff) [reply packages/template/src/index.test.ts:1]: I wanted the template to have 80% as the default threshold so i had to add this corny test.

File: .github/workflows/unit-test.yml
Comment (TheHaff) [reply .github/workflows/unit-test.yml:37]: I did it because i have no idea what I'm doing 🥇 It was a more convenient way to point to the correct output json.

File: .github/workflows/pull-request.yml
Comment (TheHaff) [reply .github/workflows/pull-request.yml]: [Invalid workflow file: .github/workflows/pull-request.yml#L12](https://github.com/tidal-music/tidal-sdk-web/actions/runs/8055171339/workflow)
The workflow is not valid. .github/workflows/pull-request.yml (Line: 12, Col: 3): Error calling workflow 'tidal-music/tidal-sdk-web/.github/workflows/unit-test.yml@469b4de9869b411e30b08c96974b0d933e884792'. The nested job 'unit-tests' is requesting 'pull-requests: write', but is only allowed 'pull-requests: none'.

Comment (TheHaff) [reply .github/workflows/pull-request.yml]: That happened when i removed it

**Review by michpohl — APPROVED** [review]
Approving so it doesn't go stale. Will try to follow up with an improved version asap.
Please wait for @osmestad and @MathiasTim to ok this decision before merging though

**Review by MathiasTim — APPROVED** [review]

File: packages/common/src/index.ts
Comment (osmestad) [inline packages/common/src/index.ts:1]: was there a reason this needed ignores?

**Review by osmestad — APPROVED** [review]

**Pull request merged**
